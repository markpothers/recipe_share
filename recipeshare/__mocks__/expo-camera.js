

export const Camera = {
	requestCameraPermissionsAsync: () => {
		return Promise.resolve({granted: true})
	}
}
