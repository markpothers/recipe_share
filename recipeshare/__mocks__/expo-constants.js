export default {
	manifest: {
		ios: {
			buildNumber: 22
		},
		android: {
			versionCode: 22
		}
	}
}
