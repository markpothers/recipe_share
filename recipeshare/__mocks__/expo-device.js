let deviceType = 1

export const setMockDeviceType = (type) => {
    deviceType = type
}

export const getDeviceTypeAsync = jest.fn(() => {
    // console.warn('Im here!')
    return new Promise(resolve => {
        resolve(deviceType)
    })
});