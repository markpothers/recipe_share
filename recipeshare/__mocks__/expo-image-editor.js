import React, { useEffect } from "react";
import { View } from "react-native";

export const ImageEditor = (props) => {
	useEffect(()=>{
		props.onEditingComplete({
			cancelled: false,
			uri: props.imageUri,
		})
	}, [props])
	return <View testID={"mockImageEditor"}></View>;
};
