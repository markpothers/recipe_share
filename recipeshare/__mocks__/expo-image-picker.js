

export const launchCameraAsync = () => {
	return Promise.resolve({
		canceled: false,
		assets: [{uri: "mockCameraUri"}]
	})
}

export const launchImageLibraryAsync = () => {
	return Promise.resolve({
		canceled: false,
		assets: [{uri: "mockLibraryUri"}]
	})
}
