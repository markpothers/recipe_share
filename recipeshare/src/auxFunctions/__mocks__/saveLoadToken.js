export function saveToken () {
	console.log("mock save token!!!!!")
	return null
}

export function loadToken () {
	return null
}
