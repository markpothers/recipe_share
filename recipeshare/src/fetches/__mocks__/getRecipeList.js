import { mockRecipeList } from '../../../__mocks__/recipeList/mockRecipeList'

export const getRecipeList = () => {
    // console.warn('mock fetch was called')
    // console.warn(mockRecipeList)
    return new Promise((resolve, reject) => {
        resolve(mockRecipeList)
    })
}
