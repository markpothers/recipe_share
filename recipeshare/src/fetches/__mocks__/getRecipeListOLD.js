import { mockRecipeList } from "../../../__mocks__/mockRecipeList"

export const getRecipeList = () => {
    console.warn("mock fetch was called")
    // console.warn(mockRecipeList)
    return new Promise((resolve, reject) => {
        resolve(mockRecipeList)
    })
}
