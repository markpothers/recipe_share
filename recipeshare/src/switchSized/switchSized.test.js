/**
 * @jest-environment jsdom
 */
import React from 'react';
import { configure, shallow, mount } from 'enzyme'
import { act } from 'react-dom/test-utils';
import {createSerializer} from 'enzyme-to-json';
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
import toJson from 'enzyme-to-json';

import { Switch } from 'react-native'
import { create } from 'react-test-renderer';
import SwitchSized from './switchSized.js'
import { setMockDeviceType } from '../../__mocks__/expo-device'

describe('SwitchSized', () => {

	let component
	let mockChangeFunction

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(async() => {
		// console.log('runs before every test')
		mockChangeFunction = jest.fn()
		// await act(async () => {
		// 	component = await mount(
		// 		<SwitchSized
		// 			value={true}
		// 			onChange={mockChangeFunction}
		// 		/>
		// 	)
		// })
		// component.setProps({})
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	test('can be rendered on an iPhone with reduced size switches', async() => {
		setMockDeviceType(1)
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered on non-iPhone with regular switch', async() => {
		setMockDeviceType(2)
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered and switched', async() => {
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let testSwitch = component.find(Switch)
		expect(testSwitch.length).toEqual(1)
		testSwitch.props().onChange()
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(1)
	})

});
