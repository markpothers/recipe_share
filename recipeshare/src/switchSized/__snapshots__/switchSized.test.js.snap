// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSized can be rendered on an iPhone with reduced size switches 1`] = `
<SwitchSized
  onChange={[MockFunction]}
  value={true}
>
  <Switch
    onChange={[MockFunction]}
    style={
      Object {
        "transform": Array [
          Object {
            "scaleX": 0.7,
          },
          Object {
            "scaleY": 0.7,
          },
        ],
      }
    }
    thumbColor="#4b7142"
    trackColor={
      Object {
        "true": "#4b714299",
      }
    }
    value={true}
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#4b714299"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 0.7,
              },
              Object {
                "scaleY": 0.7,
              },
            ],
          },
        ]
      }
      thumbTintColor="#4b7142"
      value={true}
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#4b714299"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "transform": Array [
                Object {
                  "scaleX": 0.7,
                },
                Object {
                  "scaleY": 0.7,
                },
              ],
            },
          ]
        }
        thumbTintColor="#4b7142"
        value={true}
      />
    </RCTSwitch>
  </Switch>
</SwitchSized>
`;

exports[`SwitchSized can be rendered on non-iPhone with regular switch 1`] = `
<SwitchSized
  onChange={[MockFunction]}
  value={true}
>
  <Switch
    onChange={[MockFunction]}
    style={null}
    thumbColor="#4b7142"
    trackColor={
      Object {
        "true": "#4b714299",
      }
    }
    value={true}
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#4b714299"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      thumbTintColor="#4b7142"
      value={true}
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#4b714299"
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        thumbTintColor="#4b7142"
        value={true}
      />
    </RCTSwitch>
  </Switch>
</SwitchSized>
`;
